janitor/functions/pivot.py

```diff
@@ -110,6 +109,31 @@ def pivot_longer(
         0   1        sp      m  5564      2
         1   1       rel      f    65      3
 
+    Use multiple `.value` to reshape dataframe:
+
+        >>> df = pd.DataFrame(
+        ...     [
+        ...         {
+        ...             "x_1_mean": 1,
+        ...             "x_2_mean": 1,
+        ...             "y_1_mean": 1,
+        ...             "y_2_mean": 1,
+        ...             "unit": 1,
+        ...         }
+        ...     ]
+        ... )
+        >>> df
+           x_1_mean  x_2_mean  y_1_mean  y_2_mean  unit
+        0         1         1         1         1     1
+        >>> df.pivot_longer(
+        ...     index="unit",
+        ...     names_to=(".value", "time", ".value"),
+        ...     names_pattern=r"(x|y)_([0-9])(_mean)",
+        ... )
+           unit time  x_mean  y_mean
+        0     1    1       1       1
+        1     1    2       1       1
```

Comment:
I'm thinking if we could use different numbers instead of all `1`'s to demonstrate the functionality a little clearer. Perhaps 10, 20, 30, 40, ? to distinguish from the 1,2 in the `time` column, for example.